packages/transaction-controller/package.json

```diff
@@ -52,6 +52,7 @@
     "@metamask/controller-utils": "^11.0.0",
     "@metamask/eth-query": "^4.0.0",
     "@metamask/gas-fee-controller": "^17.0.0",
+    "@metamask/keyring-api": "6.4.0",
```

Comment:
Looking at the emitted type declaration file, it looks like the `@metamask/keyring-api` shows up there, so this needs to be in `dependencies`, not `devDependencies` or `peerDependencies`.

Reply:
Unrelated, it looks like this is missing a `^`:

``` suggestion
    "@metamask/keyring-api": "^6.4.0",

Reply:
@mcmire I may be missing something but it seems to me that if we just consider the `peerDependencies` aspect in isolation, it should also be here.
And if the `peerDependencies` is satisifed, then there should be no TypeScript error on build, and therefore no need for `dependencies` since the types will be provided by the `peerDependencies` entry?

Reply:
I'm thinking if it's not added in `peerDependencies`, using code may run the risk of getting incorrectly passing types due to using the wrong version of `@metamask/keyring-api` for type-checking?

Reply:
I've removed the callback and used the messaging system instead to get the selected account. `@metamask/keyring-api` can just be a dev dependency for the tests.